lib/CodeGen/CodeGenFunction.h

```diff
@@ -3333,22 +3340,43 @@ class CodeGenFunction : public CodeGenTypeCache {
     if (CallArgTypeInfo) {
 #ifndef NDEBUG
       bool isGenericMethod = isObjCMethodWithTypeParams(CallArgTypeInfo);
+      bool isFunctionProtoType = isFunctionProtoTypeWithTypeParams(CallArgTypeInfo);
 #endif
 
       // First, use the argument types that the type info knows about
+      int N = 0;
       for (auto I = CallArgTypeInfo->param_type_begin() + ParamsToSkip,
                 E = CallArgTypeInfo->param_type_end();
-           I != E; ++I, ++Arg) {
+           I != E; ++I, ++Arg, ++N) {
         assert(Arg != ArgRange.end() && "Running over edge of argument list!");
+        const Type* ty1 = getContext().getCanonicalType((*I).getNonReferenceType()).getTypePtr();
```

Comment:
All this code should be under ifndef NDEBUG.   We only want this code to run if the assert will be executed.